### PR TITLE
UI widgets/permissions

### PR DESCRIPTION
**Pull request by ar4s**
New widget for permissions:
ralph_1547

connected to #1467

### CODE REVIEW

File: src/ralph/admin/widgets.py [hunk 1]
```diff
@@ -19,3 +23,68 @@ def media(self):
     def render(self, name, value, attrs=None):
         attrs['class'] = 'datepicker'
         return super(AdminDateWidget, self).render(name, value, attrs=attrs)
+
+
+class PermissionsSelectWidget(forms.Widget):
+    @property
+    def media(self):
+        js = ['multi.js']
+        return forms.Media(js=[static('src/js/%s' % path) for path in js])
```
Comment (mkurek) [inline src/ralph/admin/widgets.py]: os.path.join? :)

File: src/ralph/admin/widgets.py [hunk 2]
```diff
@@ -19,3 +23,68 @@ def media(self):
...
+        attrs = {'id': input_id, 'type': 'checkbox', 'value': option_value}
+        if option_value in selected_choices:
+            attrs['checked'] = 'checked'
+        attrs = self.build_attrs(**attrs)
+        return '<input{}><label{}>{}</label>'.format(
+            flatatt(attrs), flatatt({'for': input_id}), option_label
+        )
+
+    def render_options(self, choices, selected_choices):
+        self.choices = sorted(self.choices, key=lambda x: x[1].split(' | ')[1])
```
Comment (ghost) [inline src/ralph/admin/widgets.py]: i would make ' | ' const or anything else that is defined in one place and use everywhere ;)

File: src/ralph/admin/templates/ralph_admin/change_form.html [hunk 3]
```diff
@@ -17,7 +17,7 @@
   <div id="content-main">
     <div class="row header">
       <div class="large-4 columns">
-        <h1>{% sitetree_page_title from 'ralph_admin' %}</h1>
+        {# <h1>{% sitetree_page_title from 'ralph_admin' %}</h1> #}
```
Comment (xliiv) [inline src/ralph/admin/templates/ralph_admin/change_form.html]: ?

File: src/ralph/admin/widgets.py (hunk 2 above)
Comment (ghost) [inline src/ralph/admin/widgets.py]: can you give an example of choice value?

File: src/ralph/admin/widgets.py [hunk 4]
```diff
@@ -19,3 +23,68 @@ def media(self):
...
+        grouped = groupby(self.choices, lambda x: x[1].split(' | ')[1])
+        rendered_options = ''
+        for group in grouped:
+            items = list(group[1])
+            local_values = [x[0] for x in items]
+            local_selected = set(local_values) - set(selected_choices)
+            rendered_options += mark_safe(
+                '<li class="accordion-navigation">'
+                    '<a href="#{}">'
+                        '{}&nbsp;<i class="right">'
```
Comment (ghost) [inline src/ralph/admin/widgets.py]: could you add names to these placeholders for better readablity? :)

File: src/ralph/static/src/js/multi.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    $(document).ready(function() {
+        var addToCounter = function(element, value) {
+            var $counter = $('.counter', $(element).parents('.accordion-navigation'));
+            $counter.html(parseInt($counter.html()) + value);
+        };
+
+        $.each($('.accordion[data-multi]'), function(index, value) {
+            var $acordion = $(value);
+            var $input = $('#id_' + $acordion.data('multi'));
+            $acordion.data('value', $input.val().split(','));
```
Comment (ghost) [inline src/ralph/static/src/js/multi.js]: accordion?

File: src/ralph/static/src/js/multi.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            $('input[type=checkbox]', $acordion).click(function() {
```
Comment (ghost) [inline src/ralph/static/src/js/multi.js]: Consider
https://api.jquery.com/checked-selector/

File: src/ralph/admin/widgets.py [hunk 7]
```diff
@@ -19,3 +23,68 @@ def media(self):
...
+
+
+class PermissionsSelectWidget(forms.Widget):
+    @property
+    def media(self):
+        js = ['multi.js']
+        return forms.Media(js=[static('src/js/%s' % path) for path in js])
+
+    def value_from_datadict(self, data, files, name):
+        values = data.get(name, None)
```
Comment (szok) [inline src/ralph/admin/widgets.py]: Default get() return None

**Comment by szok** [comment]
:ok:

File: src/ralph/admin/widgets.py [hunk 8]
```diff
@@ -1,21 +1,104 @@
...
+
+    def render_options(self, choices, selected_choices):
+        separator = ' | '
+        self.choices = sorted(
+            self.choices, key=lambda x: x[1].split(separator)[1]
+        )
+        grouped = groupby(self.choices, lambda x: x[1].split(separator)[1])
+        rendered_options = ''
+        for group in grouped:
+            items = list(group[1])
```
Comment (mkurek) [inline src/ralph/admin/widgets.py]: could you create some pretty names for `group[0]` and `group[1]`? Ex. `group_key` and `group_choices`?

**Comment by ghost** [comment]
:ok:

**Pull request merged**